src/message/groups/trans_idx_sig_groups.rs

```diff
@@ -0,0 +1,128 @@
+use crate::error::ParsideResult;
+use crate::message::cold_code::ColdCode;
+use crate::message::controller_idx_sigs::ControllerIdxSig;
+use crate::message::groups::controller_idx_sigs::ControllerIdxSigs;
+use crate::message::parsers::Parsers;
+use crate::message::{Group, GroupItem};
+use cesride::counter::Codex;
+use cesride::{Counter, Matter, Prefixer, Saider, Seqner};
+use nom::multi::count;
+use nom::sequence::tuple;
+
+#[derive(Debug, Clone, Default)]
+pub struct TransIdxSigGroups {
+    pub value: Vec<TransIdxSigGroup>,
+}
+
+impl Group<TransIdxSigGroup> for TransIdxSigGroups {
+    const CODE: &'static str = Codex::TransIdxSigGroups;
+
+    fn new(value: Vec<TransIdxSigGroup>) -> Self {
+        Self { value }
+    }
+
+    fn value(&self) -> &Vec<TransIdxSigGroup> {
+        &self.value
+    }
+}
+
+impl TransIdxSigGroups {
+    pub(crate) fn from_stream_bytes<'a>(
+        bytes: &'a [u8],
+        counter: &Counter,
+        cold_code: &ColdCode,
+    ) -> ParsideResult<(&'a [u8], TransIdxSigGroups)> {
+        let (rest, body) = count(
+            tuple((
+                Parsers::prefixer_parser(cold_code)?,
+                Parsers::seqner_parser(cold_code)?,
+                Parsers::saider_parser(cold_code)?,
+                Parsers::siger_list_parser(cold_code)?,
+            )),
+            counter.count() as usize,
+        )(bytes)?;
+
+        let body = body
+            .into_iter()
+            .map(|(prefixer, seqner, saider, isigers)| TransIdxSigGroup {
+                prefixer,
+                seqner,
+                saider,
+                isigers: ControllerIdxSigs::new(
+                    isigers.into_iter().map(ControllerIdxSig::new).collect(),
+                ),
+            })
+            .collect();
+
+        Ok((rest, TransIdxSigGroups { value: body }))
+    }
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct TransIdxSigGroup {
+    pub prefixer: Prefixer,
+    pub seqner: Seqner,
+    pub saider: Saider,
+    pub isigers: ControllerIdxSigs,
+}
+
+impl TransIdxSigGroup {
+    pub fn new(
+        prefixer: Prefixer,
+        seqner: Seqner,
+        saider: Saider,
+        isigers: ControllerIdxSigs,
+    ) -> Self {
+        Self { prefixer, seqner, saider, isigers }
+    }
+}
+
+impl GroupItem for TransIdxSigGroup {
+    fn qb64(&self) -> ParsideResult<String> {
+        let mut out = String::new();
+        out.push_str(&self.prefixer.qb64()?);
+        out.push_str(&self.seqner.qb64()?);
+        out.push_str(&self.saider.qb64()?);
+        out.push_str(&self.isigers.qb64()?);
+        Ok(out)
+    }
+
+    fn qb64b(&self) -> ParsideResult<Vec<u8>> {
+        let mut out = Vec::new();
+        out.extend_from_slice(&self.seqner.qb64b()?);
+        out.extend_from_slice(&self.seqner.qb64b()?);
+        out.extend_from_slice(&self.saider.qb64b()?);
+        out.extend_from_slice(&self.isigers.qb64b()?);
```

Comment:
```suggestion
        let capacity = self.prefixer.full_size() + self.seqner.full_size() + self.saider.full_size() + self.isigers.full_size();
        let mut out = vec![0u8; capacity];
        let mut offset = 0usize;
        let mut length = self.prefixer.full_size() as usize;
        out[offset..(offset + length)].copy_from_slice(&self.prefixer.qb64b()?);
        offset += length;
        length = self.seqner.full_size() as usize;
        out[offset..(offset + length)].copy_from_slice(&self.seqner.qb64b()?);
        offset += length;
        length = self.saider.full_size() as usize;
        out[offset..(offset + length)].copy_from_slice(&self.saider.qb64b()?);
        offset += length;
        length = self.isigers.full_size() as usize;
        out[offset..(offset + length)].copy_from_slice(&self.isigers.qb64b()?);
```

I think this would perform better - but more importantly, you have `seqner` as the first primitive here and it should be `prefixer`.

Reply:
@jasoncolburne done.